book_extractors/common/extractors/base_extractor.py

```diff
@@ -6,11 +6,19 @@
 class BaseExtractor:
     __metaclass__ = ABCMeta
 
-    def __init__(self, key_of_cursor_location_dependent=None, options=None):
+    def __init__(self, dependencies_contexts, key_of_cursor_location_dependent=None, options=None, dependencies=None):
```

Comment:
Hmm. onko tässä jokin syy välittää nämä `dependencies`:it initissä tänne? Eikös nämä ole niitä extractorissa itsessään julistettuja riippuvuuksia? Eivätkö ne voisi olla vain BaseExtractorin self-muuttujassa, jonne nuo periytetyt extractorit voivat omansa laittaa suoraan? 

Tietysti tällä voi periaatteessa välittää nuo riippuvuudet ulkopuolelta tarvittaessa, oliko se ajatuksena ja jos oli niin oliko sulla mielessä jokin tilanne sille?